experimental/dds/tree2/src/test/feature-libraries/default-field-kinds/defaultChangeFamily.spec.ts

```diff
@@ -301,62 +315,139 @@ describe("DefaultEditBuilder", () => {
 	});
 
 	describe("Sequence Field Edits", () => {
-		it("Can insert a root node", () => {
-			const { builder, forest } = initializeEditableForest();
-			builder
-				.sequenceField({ parent: undefined, field: rootKey })
-				.insert(0, cursorForJsonableTreeNode(nodeX));
-			expectForest(forest, nodeX);
-		});
+		describe("Can insert a root node", () => {
+			it("with uncompressed encoding", () => {
+				const { builder, forest } = initializeEditableForest();
+				builder
+					.sequenceField({ parent: undefined, field: rootKey })
+					.insert(0, cursorForJsonableTreeNode(nodeX));
+				expectForest(forest, nodeX);
+			});
 
-		it("Can insert a child node", () => {
-			const { builder, forest } = initializeEditableForest({
-				type: jsonObject.name,
-				fields: {
-					foo: [
-						{ type: leaf.number.name, value: 0 },
-						{ type: leaf.number.name, value: 1 },
+			it("with schema based encoding", () => {
+				const { builder, forest } = initializeEditableForest();
+				builder
+					.sequenceField(
+						{ parent: undefined, field: rootKey },
 						{
-							type: jsonObject.name,
-							fields: {
-								foo: [
-									{ type: leaf.number.name, value: 0 },
-									{ type: leaf.number.name, value: 1 },
-									{ type: leaf.number.name, value: 2 },
-									{ type: leaf.number.name, value: 3 },
-									{ type: leaf.number.name, value: 4 },
-								],
-							},
+							schema: buildSchema(
+								TreeFieldSchema.create(FieldKinds.required, [leaf.string]),
+							),
+							policy: defaultSchemaPolicy,
 						},
-					],
-				},
+					)
+					.insert(0, cursorForJsonableTreeNode(nodeX));
+				expectForest(forest, nodeX);
 			});
-			builder
-				.sequenceField({ parent: root_foo2, field: fooKey })
-				.insert(5, cursorForJsonableTreeNode(nodeX));
-			const expected = {
-				type: jsonObject.name,
-				fields: {
-					foo: [
-						{ type: leaf.number.name, value: 0 },
-						{ type: leaf.number.name, value: 1 },
-						{
-							type: jsonObject.name,
-							fields: {
-								foo: [
-									{ type: leaf.number.name, value: 0 },
-									{ type: leaf.number.name, value: 1 },
-									{ type: leaf.number.name, value: 2 },
-									{ type: leaf.number.name, value: 3 },
-									{ type: leaf.number.name, value: 4 },
-									nodeX,
-								],
+		});
+
+		describe("Can insert a child node", () => {
+			it("with uncompressed encoding", () => {
```

Comment:
I'm a bit confused with this test.

It seems like its initializing the forest to a somewhat large and needlessly complex tree (thats 9 nodes I think), then doing the insert you want to test, then comparing it to what looks like the same tree?

Why is the tree so big?
Why are you duplicating the source code for the jsonable tree?
Why is the expected value and the initial value the same? I thought this was testing the insertion operation which should change the tree?
If we are testing the encoding of ops produced by the builder, can we do that without involving a forest at all, and just make an edit builder, and have it encode an op?

Reply:
The tests were just duplicates from the existing tests for sequenceField insert tests in the file, but with the shapeInfo passed through (to use schema based encoding). The expected and initial values were not quite the same (as the expected now has the added node `nodeX` in it).

But yes making the tests simpler with just the edit builder makes much more sense!

I've updated the test file to test for schema based encoding on the testTrees used for the `schemaBasedEncoding.spec.ts` using just an edit builder instead.

Reply:
Thats a nice idea. I like the new test setup!